src/lib/addons/slack-app.ts

```diff
@@ -17,64 +21,103 @@ interface ISlackAppAddonParameters {
-            const featureLink = this.msgFormatter.featureLink(event);
-
-            const requests = slackChannelsToPostTo.map(({ id }) =>
-                this.slackClient!.chat.postMessage({
-                    channel: id!,
-                    text,
-                    attachments: [
-                        {
-                            actions: [
-                                {
-                                    name: 'featureToggle',
-                                    text: 'Open in Unleash',
-                                    type: 'button',
-                                    value: 'featureToggle',
-                                    style: 'primary',
-                                    url: featureLink,
-                                },
-                            ],
-                        },
-                    ],
-                }),
-            );
-
-            await Promise.all(requests);
-            this.logger.info(`Handled event ${event.type}.`);
+                const text = this.msgFormatter.format(event);
+                const featureLink = this.msgFormatter.featureLink(event);
+
+                const requests = slackChannelsToPostTo.map(({ id }) =>
+                    this.slackClient!.chat.postMessage({
+                        channel: id!,
+                        text,
+                        attachments: [
+                            {
+                                actions: [
+                                    {
+                                        name: 'featureToggle',
+                                        text: 'Open in Unleash',
+                                        type: 'button',
+                                        value: 'featureToggle',
+                                        style: 'primary',
+                                        url: featureLink,
+                                    },
+                                ],
+                            },
+                        ],
+                    }),
+                );
+
+                await Promise.all(requests);
+                this.logger.info(`Handled event ${event.type}.`);
+            }
+        } catch (error) {
+            if (error.code === ErrorCode.PlatformError) {
+                this.logger.error(
+                    `Error handling event ${event.type}. A platform error occurred: ${error.data}`,
+                    error,
+                );
+            } else if (error.code === ErrorCode.RequestError) {
+                this.logger.error(
+                    `Error handling event ${event.type}. A request error occurred: ${error.original}`,
+                    error,
+                );
+            } else if (error.code === ErrorCode.RateLimitedError) {
+                this.logger.error(
+                    `Error handling event ${event.type}. A rate limit error occurred: retry after ${error.retryAfter} seconds`,
+                    error,
+                );
```

Comment:
IIRC, WebClient will handle all these for us, right? If that's the case, should the log be an error or just a warn? Maybe it could be an info message saying that "It will be retried" more than "Retry later" which seems to indicate the client/user has to do something

Reply:
In this case, this is still an error and should only happen in case `WebClient` does not handle rate limiting for us.
https://slack.dev/node-slack-sdk/web-api#rate-limits
>You might not want to the WebClient to handle rate limits in this way. Perhaps the operation was time sensitive, and it won’t be useful by the time Slack is ready for another request. Or, you have a more sophisticated approach. In these cases, you can set the rejectRateLimitedCalls option on the client to true. Once you set this option, method calls can fail with rate limiting related errors. These errors have a code property set to ErrorCode.RateLimitedError. See [error handling](https://slack.dev/node-slack-sdk/web-api#handle-errors) for more details.

I've added a callback for the `RATE_LIMITED` event so we can log it. I'm keeping this error handling for now, but we can remove it if you think it's better.

Reply:
Yeah, I wasn't concerned about logging when `WebClient` receives a rate limit and handles it internally, I was rather thinking that the error might not happen.

It's ok what you did in https://github.com/Unleash/unleash/commit/1d2db10de70b3c2b1cdb3bda3df57cb9b6683131 I'm not sure if `client.on(WebClientEvent.RATE_LIMITED` is needed (or if it has to be a warn, could be maybe a debug), but we can change this later, so no worries for now

Reply:
Ah yeah, fair point. I'll keep it for now then, but I changed it to `debug` for now: https://github.com/Unleash/unleash/pull/4284/commits/a8a7410209bcc74480b03c27c0cf0951c9e457f1